src/dispatch/scheduler.py

```diff
@@ -158,13 +161,46 @@ class OneShotScheduler:
     take over scheduling asynchronous calls.
     """
 
-    __slots__ = ("entry_point", "version", "poll_max_wait_seconds")
+    __slots__ = (
+        "entry_point",
+        "version",
+        "poll_min_results",
+        "poll_max_results",
+        "poll_max_wait_seconds",
+    )
 
     def __init__(
-        self, entry_point: Callable, version=sys.version, poll_max_wait_seconds=5
+        self,
+        entry_point: Callable,
+        version: str = sys.version,
+        poll_min_results: int = 1,
+        poll_max_results: int = 10,
+        poll_max_wait_seconds: int = 60,
```

Comment:
Should we disable the max wait by default? If we're waiting on a call that's retrying for a couple of hours, there's little value in waking up every minutes isn't it?

Reply:
To me this parameter is only useful to manage internally to integrate with timers. Do you have another use case in mind for forcing the coroutine to wake up?

Reply:
See https://github.com/stealthrocket/dispatch-sdk-python/pull/92/commits/1269929304d9590074688ad530382fb263edcc8e.

We'll need to be careful if/when we introduce https://github.com/stealthrocket/dispatch-sdk-python/issues/89 so that we don't yield to Dispatch with `max_wait=None` when there are no outstanding calls, otherwise the task will stay suspended until it expires. Agreed though that there's no value in waking up otherwise.